### PR TITLE
feat: generate label dimensions for IRI dimensions

### PR DESCRIPTION
**Pull request by tpluscode**
This prepares a view in a way that any dimension which is `sh:nodeKind sh:IRI` will create additional columns for `schema:name` and `schema:termCode`

The `schema:name` column is marked as `view:labelFor`, linking to the actual concept column

A sequence of `view:as` is added so that every dimension has a unique value

### CODE REVIEW

File: packages/view-util/test/__snapshots__/index.snap [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
+
+exports["@view-builder/view-util prepareViewPointer IRI dimension with multiple sources generates a join for schema:name when dimension is sh:IRI"] = "SELECT DISTINCT ?dimension1 WHERE {\n  <https://ld.stadt-zuerich.ch/statistics/000003> <https://cube.link/observationSet> ?observationSet0 .\n  ?observationSet0 <https://cube.link/observation> ?source0 .\n  <https://ld.stadt-zuerich.ch/statistics/000015> <https://cube.link/observationSet> ?observationSet1 .\n  ?observationSet1 <https://cube.link/observation> ?source1 .\n  ?source0 <https://ld.stadt-zuerich.ch/statistics/property/RAUM> ?dimension0 .\n  ?source1 <https://ld.stadt-zuerich.ch/statistics/property/RAUM> ?dimension0 .\n  ?dimension0 <http://schema.org/name> ?dimension1 .\n}\nGROUP BY ?dimension1";
+
```
Comment (tpluscode) [inline packages/view-util/test/__snapshots__/index.snap:6]: Re https://github.com/tlameiras/chai-snapshot-matcher/pull/29

File: packages/view-util/lib/projection.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    function nextColumnProperty() {
+      i += 1
+      return view.namedNode(`${view.value}#column${i}`)
+    }
+
+    for (const { dimension, isIri } of dimensions) {
+      dimension.addOut(ns.view.as, nextColumnProperty())
+      yield dimension
+
+      if (isIri) {
```
Comment (cristianvasquez) [inline packages/view-util/lib/projection.js:28]: I struggled to understand the logic here. Perhaps `isIri` could be named (as an alias in this function) something like `needsLabelDimension`?

File: packages/view-util/lib/projection.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        const labelDimension = view.blankNode()
+
+        labelDimension
+          .addOut(ns.view.as, nextColumnProperty())
+          .addOut(ns.view.labelFor, dimension)
```
Comment (cristianvasquez) [inline packages/view-util/lib/projection.js:33]: I don't know if `view:labelFor` is the best name; I wonder if `lookupTarget` would be more precise.

File: packages/view-util/lib/projection.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            from.addOut(ns.view.join, dimension)
+          })
+
+        yield labelDimension
+
+        const termCodeDimension = view.blankNode()
+
+        termCodeDimension
+          .addOut(ns.view.as, nextColumnProperty())
+          .addOut(ns.view.from, (from) => {
```
Comment (cristianvasquez) [inline packages/view-util/lib/projection.js:48]: never mind :)

File: packages/view-util/lib/projection.js (hunk 2 above)
Comment (tpluscode) [reply packages/view-util/lib/projection.js:28]: Sure, that's a neat idea!

File: packages/view-util/test/index.test.js [hunk 5]
```diff
@@ -3,9 +3,18 @@ import { ssz, testData } from '@view-builder/testing'
...
 describe('@view-builder/view-util', () => {
+  let cubeLookup
+
+  beforeEach(() => {
+    cubeLookup = sinon.createStubInstance(CubeLookup)
+    cubeLookup.isIriDimension.resolves(false)
+  })
+
   it('turns cube source to blank nodes', async () => {
     // given
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply packages/view-util/test/index.test.js:19]: You're commenting on unchanged code? I don't understand

**Pull request merged**